File: test/javascript/boards.js
const assert = require('assert');
const PinterestSdk = require('pinterest-sdk');

describe('boards', function() {
  describe('list pins', function() {
    it('should respond with error 401 unauthorized when token is invalid', function(done) {
      var callback = function(error, data, response) {
        assert.equal(error.message, 'Unauthorized');
        assert.equal(typeof data, 'object');
        assert.equal(response.res.statusCode, 401);
        done();
      };

      const apiClient = PinterestSdk.ApiClient.instance;
      var pinterestOAuth2 = apiClient.authentications['pinterest_oauth2'];
      pinterestOAuth2.accessToken = "SOME INVALID TOKEN"
      
      const boardsApi = new PinterestSdk.BoardsApi();
      boardsApi.boardsListPins('12345', {}, callback);
    });
  });
});

File: test/python/boards.py
import unittest
import pinterestsdk
from pinterestsdk.rest import ApiException
from pprint import pprint

class TestPinterestSdkBoards(unittest.TestCase):

    def test_boards_list_pins_unauthorized(self):
        """Test boards_list_pins API call with invalid token should fail with 401 Unauthorized"""

        configuration = pinterestsdk.Configuration(
            host = "https://api.pinterest.com/v5",
            access_token = "SOME INVALID TOKEN"
        )

        with pinterestsdk.ApiClient(configuration) as api_client:

            api_instance = pinterestsdk.BoardsApi(api_client)
            board_id = '12345'

            try:
                api_response = api_instance.boards_list_pins(board_id)
                self.fail('API called successfully when it should have failed')
            except ApiException as e:
                print("Exception when calling BoardsApi->boards_list_pins: %s\n" % e)
                self.assertEqual(e.status, 401)

if __name__ == '__main__':
    unittest.main()
